stack/moving-to-github-actions.rst

```diff
@@ -0,0 +1,76 @@
+####################################
+Moving to GitHub Actions from Travis
+####################################
+
+GitHub Actions execute faster and more reliably than Travis scripts, so we are converting existing repositories to use them.
+This document describes how to accomplish this transition for existing packages.
+New packages should start with GitHub Actions by default; additional customizations that may be useful are suggested in :doc:`the new package documentation</stack/adding-a-new-package>`.
+
+Travis uses a ``.travis.yml`` file that specifies the workflow to be executed when events such as pushes or pull requests occur for the repository.
+GitHub Actions use ``.github/workflows/*.yaml`` files instead.
+These workflows execute in parallel.
+Migrating a package consists of replacing the ``.travis.yml`` file with one (or more) appropriate GitHub Actions workflows files.
+
+If the ``.travis.yml`` file contains only the default execution of ``flake8`` for Python "lint" checking, this can be done entirely in :ref:`the GitHub web interface <github-actions-web-interface>`, or it can be done a bit more cleanly from :ref:`the command line <github-actions-command-line>`.
+
+If your ``.travis.yml`` does more than run ``flake8``, please consult an expert for assistance.
+
+
+.. _github-actions-web-interface:
+
+GitHub web interface
+====================
+
+#. Go to the repository you are transitioning at ``https://github.com/lsst/{repo}``.
+#. Select ``.travis.yml``
+#. Select the "trash can" icon to delete this file.
+#. Add a commit comment such as "Remove Travis workflow." and choose a branch name.
+   This branch name doesn't need to follow our ``tickets/`` convention; ``u/{user}/remove-travis`` is adequate.
+#. Commit the change, create a pull request, and merge it.
+#. Select the "Actions" tab between "Pull Requests" and "Projects" at the top of the GitHub page.
+#. Choose the "LSST DM Python lint Workflow" and click on "Set up this workflow".
+#. Choose "Start commit".
+#. Add a commit comment such as "Add Python lint workflow." and choose a branch name.
+   Again, something like ``u/{user}/add-lint-workflow`` is sufficient as a branch name.
+#. Commit the change, create a pull request, and merge it.
+#. Update the repository settings using :ref:`the procedure below <github-actions-settings>`.
+
+
+.. _github-actions-command-line:
+
+Command line
+============
+
+#. Clone the repository you are transitioning from ``git@github.com:lsst/{repo}``.
+#. Create a branch: ``git checkout -b u/{user}/migrate-to-gha``
+#. ``git rm .travis.yml``
+#. ``mkdir -p .github/workflows; cd .github/workflows``
+#. ``curl -LO https://lsst/.github/template-workflows/lint.yaml``
```

Comment:
Should this be a link to the templates repository?

Reply:
It is unfortunate that there are multiple sources of truth, but I think I'd prefer that `templates` grab from `.github` than the other way around.

Reply:
Ah, ideally the stack_package template's SConscript file should get downloading the `link.yaml` from the `.github` repository.

But more to the point, this URL is missing the domain? Should be https://raw.githubusercontent.com/lsst/.github/master/workflow-templates/lint.yaml ?

Reply:
Doh.  Yes, thanks for catching that.